synapse/cortex.py

```diff
@@ -2495,6 +2499,52 @@ async def _initCoreLayers(self):
             layrinfo = await node.dict()
             await self._initLayr(layrinfo)
 
+    async def cloneLayer(self, iden):
+        '''
+        Make a copy of a Layer in the cortex.
+
+        Args:
+            iden (str): layer iden
+        '''
+        layr = self.layers.get(iden, None)
+        if layr is None:
+            raise s_exc.NoSuchLayer(iden=iden)
+
+        newiden = s_common.guid()
+
+        return await self._push('layer:clone', iden, newiden)
+
+    @s_nexus.Pusher.onPush('layer:clone')
+    async def _cloneLayer(self, iden, newiden):
+
+        if newiden in self.layers:
+            return
+
+        layr = self.layers.get(iden)
+        if layr is None:
+            return
+
+        newpath = s_common.gendir(self.dirn, 'layers', newiden)
+        await layr.clone(newpath)
+
+        node = await self.hive.open(('cortex', 'layers', iden))
+        copynode = await self.hive.open(('cortex', 'layers', newiden))
+
+        layrinfo = await node.dict()
+        copyinfo = await copynode.dict()
+        for name, valu in layrinfo.items():
+            await copyinfo.set(name, valu)
```

Comment:
Technically, there's a small race here (since someone else might be manipulating the layer here). Not sure if it is worth worrying about.

Reply:
I think we can let that one go